### PR TITLE
Update Unity to 2021.3.15f1, add loading riddles from Firebase

### PR DESCRIPTION
**Pull request by travis-bittel**
Issues when building for Android on our previous Unity version prompted the switch to the long-term support version. Android builds seem to work there.

### CODE REVIEW

File: Gaming The Syllabus/Assets/DebugTextManager.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    private void Awake()
+    {
+        InitializeSingleton();
+    }
+
+    public static void Log(string text)
+    {
+        _instance.textElement.text = text;
+    }
+}
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/DebugTextManager.cs:19]: I'm assuming this is for making debugging easier and cleaner. Looks good!

File: Gaming The Syllabus/Assets/Scripts/Riddles/SyllabusRiddleManager.cs [hunk 2]
```diff
@@ -50,8 +50,13 @@ private static void LoadRiddlesFromXML(string filePathToRiddlesFolder)
...
             // We end up with the Assets folder in the path twice since our XmlUtilities class uses the app data path which
             // is the path to the Assets folder, so we just remove it here.
-            riddles.Add(XmlUtilities.Deserialize<Riddle>(riddlePath.Replace("Assets/", "")));
+            Riddle riddle = XmlUtilities.DeserializeFromAbsolutePath<Riddle>(riddlePath);
+            riddles.Add(riddle);
         }
+
+        Debug.Log(riddles.Count);
+
+        SyllabusRiddleUIManager.DisplayRiddle(riddles[0]);
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/Riddles/SyllabusRiddleManager.cs:59]: I know this is for testing purposes only, but we definitely need to make sure that the UI for riddles isn't automatically displayed on start in the final build.

File: Gaming The Syllabus/Assets/Scripts/XmlUtilities.cs [hunk 3]
```diff
@@ -16,15 +16,30 @@ public static void Serialize(object item, string filePath)
 
     public static T Deserialize<T>(string filePath)
     {
+        Debug.Log(filePath);
+
         XmlSerializer serializer = new XmlSerializer(typeof(T));
         // Worth remembering that Path.Combine will just return the 2nd filePath if it is absolute (starts with a /)
-        StreamReader reader = new StreamReader(Path.Combine(Application.dataPath, filePath));
+        StreamReader reader = new StreamReader(Path.Combine(Application.persistentDataPath, filePath));
```
Comment (MatthewEpperson) [inline Gaming The Syllabus/Assets/Scripts/XmlUtilities.cs:23]: Good call on using persistentDataPath to ensure important data such as user progress is saved in between play sessions.

File: Gaming The Syllabus/Assets/DebugTextManager.cs (hunk 1 above)
Comment (travis-bittel) [reply Gaming The Syllabus/Assets/DebugTextManager.cs:19]: Not just easier and cleaner but possible really :( 
Android doesn't have good debugging support so this seems like the best I could do...

File: Gaming The Syllabus/Assets/Scripts/Riddles/SyllabusRiddleManager.cs (hunk 2 above)
Comment (travis-bittel) [reply Gaming The Syllabus/Assets/Scripts/Riddles/SyllabusRiddleManager.cs:59]: Good catch, this was changed in the latest PR for this sprint!

**Pull request merged**